File: messaging-core/src/main/java/org/torquebox/messaging/core/Topic.java
package org.torquebox.messaging.core;

public class Topic extends Destination {
	
	public Topic() {
		
	}
	
	public void start() throws Exception {
		log.info( "start topic " + getName() );
		
		getServer().createTopic(false, getName() );
	}
	
	public void destroy() throws Exception {
		log.info( "destroy topic " + getName() );
		
		getServer().destroyTopic( getName() );
	}

}

File: gems/messaging/lib/torquebox/messaging/core/topic.rb
require 'torquebox/messaging/core/destination'

module TorqueBox
  module Messaging
    module Core
      class Topic < Destination
        def initialize(jms_topic)
          super
        end

        def to_s
          "[Topic: #{super}]"
        end
      end
    end
  end
end
